sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -478,17 +478,26 @@ case class MakeYMInterval(years: Expression, months: Expression)
   override def dataType: DataType = YearMonthIntervalType()
 
   override def nullSafeEval(year: Any, month: Any): Any = {
-    Math.toIntExact(Math.addExact(month.asInstanceOf[Number].longValue(),
-      Math.multiplyExact(year.asInstanceOf[Number].longValue(), MONTHS_PER_YEAR)))
+    try {
+      Math.toIntExact(Math.addExact(month.asInstanceOf[Number].longValue(),
+        Math.multiplyExact(year.asInstanceOf[Number].longValue(), MONTHS_PER_YEAR)))
```

Comment:
I'll move it to separate function, so there will be no need to catch exception in `doGenCode` part.

Reply:
No strong preferences about this on my side.

Reply:
I've tried, but failed to `"File 'generated.java', Line 37, Column 68: Instance method "makeYearMonthInterval" cannot be invoked in static context"`, so I put this idea aside.